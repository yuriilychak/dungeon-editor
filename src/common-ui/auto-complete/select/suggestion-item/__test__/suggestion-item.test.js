import React from "react";
import { createMount } from "@material-ui/core/test-utils";

import SuggestionItem from "../suggestion-item";

describe("SuggestionItem test",()=> {
    const defaultProps = {
        isHighlighted: false,
        isSelected: false,
        suggestion: { item: "test" },
        getItemProps: (suggestion) => ({})
    };

    it("Default snapshot", () => {
        const wrapper = createMount()(
            <SuggestionItem
                {...defaultProps}
            />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Highlighted snapshot", () => {
        const wrapper = createMount()(
            <SuggestionItem
                {...defaultProps}
                isHighlighted
            />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Selected snapshot", () => {
        const wrapper = createMount()(
            <SuggestionItem
                {...defaultProps}
                isSelected
            />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });
});
