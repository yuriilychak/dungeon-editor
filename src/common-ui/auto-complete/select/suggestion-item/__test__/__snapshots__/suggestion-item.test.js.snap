// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionItem test Default snapshot 1`] = `"<div class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-1 makeStyles-menuItemDefault-3 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">test<span class=\\"MuiTouchRipple-root\\"></span></div>"`;

exports[`SuggestionItem test Highlighted snapshot 1`] = `"<div class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-1 makeStyles-menuItemDefault-3 Mui-selected MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button Mui-selected\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">test<span class=\\"MuiTouchRipple-root\\"></span></div>"`;

exports[`SuggestionItem test Selected snapshot 1`] = `"<div class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-1 makeStyles-menuItemSelected-2 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">test<span class=\\"MuiTouchRipple-root\\"></span></div>"`;
